include/swift/ABI/Metadata.h

```diff
@@ -3316,10 +3316,43 @@ struct TargetInPlaceValueMetadataCache {
 using InPlaceValueMetadataCache =
   TargetInPlaceValueMetadataCache<InProcess>;
 
+template <typename Runtime>
+struct TargetResilientClassMetadataPattern;
+
 /// An instantiation pattern for non-generic resilient class metadata.
 /// Used in conjunction with InPlaceValueMetadataInitialization.
 using MetadataRelocator =
-  Metadata *(const TargetTypeContextDescriptor<InProcess> *description);
+  Metadata *(const TargetTypeContextDescriptor<InProcess> *type,
+             const TargetResilientClassMetadataPattern<InProcess> *pattern);
+
+/// An instantiation pattern for non-generic resilient class metadata.
+template <typename Runtime>
+struct TargetResilientClassMetadataPattern {
+  /// If the class descriptor's hasResilientSuperclass() flag is set,
+  /// this field instead points at a function that allocates metadata
+  /// with the correct size at runtime.
+  TargetRelativeDirectPointer<Runtime, MetadataRelocator> RelocationFunction;
```

Comment:
This comment needs to be rewritten, since the "if" clause is now the precondition for this entire structure existing.